### ISSUE TITLE
new stacking statistics cause crash

### ISSUE THREAD

**Issue by hvasbath** [emolch/pyrocko#108]
Although I have no unreasonable input I think?

Can it happen that there are base_seismograms that contain no stacked traces so that the initial None is not being overwritten?

```yaml
--- !pf.RectangularSource
lat: 29.07
lon: 34.73
north_shift: 594.3543310463305
east_shift: -5418.356534941005
depth: 10228.371832465202
time: 1995-11-22 04:15:25.030000
stf: !pf.HalfSinusoidSTF
  duration: 17.68
  anchor: -1.0
stf_mode: post
magnitude: 6.0
strike: 186.2
dip: 61.1
rake: 81.98
length: 41250.0
width: 22590.0
velocity: 3500.0
slip: 6.57

--- !pf.RectangularSource
lat: 29.07
lon: 34.73
north_shift: -2453.746903610078
east_shift: -1180.9130546833767
depth: 6490.754529990692
time: 1995-11-22 04:15:15.880000
stf: !pf.HalfSinusoidSTF
  duration: 1.277
  anchor: -1.0
stf_mode: post
magnitude: 6.0
strike: 115.8
dip: 56.6
rake: -130.39
length: 1500.0
width: 9800.0
velocity: 3500.0
slip: 4
```
```python
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
/home/vasyurhm/Software/Seismosoft/beat/src/test_seis_forward.py in <module>()
     98 synth_traces = heart.seis_synthetics(engine, sources, targets, 
     99                                      arrival_taper, filterer,
--> 100                                      plot=plot)
    101 
    102 #trace.snuffle(synth_traces)

/usr/local/lib/python2.7/dist-packages/beat/heart.pyc in seis_synthetics(engine, sources, targets, arrival_taper, filterer, reference_taperer, plot, nprocs)
    829 
    830     response = engine.process(sources=sources,
--> 831                               targets=targets, nprocs=nprocs)
    832 
    833     synt_trcs = []

/usr/local/lib/python2.7/dist-packages/pyrocko/gf/seismosizer.pyc in process(self, *args, **kwargs)
   2644                     dsource_cache={}),
   2645 
-> 2646                 nprocs=nprocs):
   2647 
   2648             tcounters_list.append(num.diff(tcounters))

/usr/local/lib/python2.7/dist-packages/pyrocko/parimap.pyc in parimap(function, *iterables, **kwargs)
     44         while True:
     45             args = [next(it) for it in iterables]
---> 46             yield function(*args, **kwargs)
     47 
     48         return

/usr/local/lib/python2.7/dist-packages/pyrocko/gf/seismosizer.pyc in process_subrequest(work, pshared)
   2309     t_stack = 0.0
   2310     for _, tr in base_seismogram.iteritems():
-> 2311         n_records_stacked += tr.n_records_stacked
   2312         t_optimize += tr.t_optimize
   2313         t_stack += tr.t_stack

TypeError: unsupported operand type(s) for +=: 'int' and 'NoneType'
```

**Comment by hvasbath** [comment]
Printed n_records_stacked:
Apparently make_same_span throws out some of the traces as it is clear that they have been actually stacked according to the numbers.
It threw out 4062 for the components.
@emolch is that ok this way?

4062
tr_stack 4062
4062
tr_stack 4062
2708
tr_stack 2708
Nach make_same_span
statistics_stack_crash n_records_st 0.0
statistics_stack_crash n_records_st 2708
statistics_stack_crash n_records_st 0.0

**Comment by hvasbath** [comment]
PR like this https://github.com/emolch/pyrocko/pull/109

**Comment by miili** [comment]
Resolved through #109
